cli/src/commands/prev.rs

```diff
@@ -78,25 +77,17 @@ pub(crate) fn cmd_prev(
             .heads()
             .contains(current_wc_id);
     let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
-    let start_id = if edit {
-        current_wc_id
+    let offset = if current_wc.is_discardable() && !args.edit {
+        // If the current working copy is discardable and the user didn't
+        // specify --edit, then we move one commit further back. This is so that
+        // the new empty working commit is created on the parent of the current
+        // on (or further back depending on the value of args.amount). If we
+        // didn't do this, then we wouldn't move far enough.
+        args.offset + 1
     } else {
-        match current_wc.parent_ids() {
-            [parent_id] => parent_id,
-            _ => return Err(user_error("Cannot run `jj prev` on a merge commit")),
-        }
-    };
-    let ancestor_expression = RevsetExpression::commit(start_id.clone()).ancestors_at(offset);
-    let target_revset = if edit {
-        ancestor_expression
-    } else {
-        // Jujutsu will always create a new commit for prev, even where Mercurial cannot
-        // and fails. The decision and all discussion around it are available
-        // here: https://github.com/martinvonz/jj/pull/1200#discussion_r1298623933
-        //
-        // If users ever request erroring out, add `.ancestors()` to the revset below.
-        ancestor_expression.minus(&RevsetExpression::commit(current_wc_id.clone()))
+        args.offset
     };
```

Comment:
???

Reply:
I changed this entire PR. Please take another look and read the last comment I left.